cmd/payments/main.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/spf13/cobra"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/provider"
+	"storj.io/storj/pkg/transport"
+)
+
+var (
+	ctx = context.Background()
+	// ErrPaymentsDial throws when there are errors dialing the payments client
+	ErrPaymentsDial = errs.Class("error dialing payments client")
+
+	// ErrRequest is for gRPC request errors after dialing
+	ErrRequest = errs.Class("error processing request")
+
+	// ErrIdentity is for errors during identity creation for this CLI
+	ErrIdentity = errs.Class("error creating identity")
+
+	// ErrArgs throws when there are errors with CLI args
+	ErrArgs = errs.Class("error with CLI args")
+
+	port string
+
+	rootCmd = &cobra.Command{Use: "payments"}
+
+	cmdGenerate = &cobra.Command{
+		Use:   "generatecsv",
+		Short: "generates payment csv",
+		Args:  cobra.MinimumNArgs(2),
+		RunE:  generatecsv,
+	}
+
+	cmdTest = &cobra.Command{
+		Use:   "test",
+		Short: "test only: add records to accounting rollup",
+		RunE:  test,
+	}
+)
+
+// Payments gives access to the payments api
+type Payments struct {
+	client pb.PaymentsClient
+}
+
+func main() {
+	rootCmd.PersistentFlags().StringVarP(&port, "port", "p", ":10000", "storj-sim satellite port")
+	rootCmd.AddCommand(cmdGenerate)
+	rootCmd.AddCommand(cmdTest)
+	err := rootCmd.Execute()
+	if err != nil {
+		fmt.Println("Error", err)
+		os.Exit(1)
+	}
+}
+
+// NewPayments creates a payments object
+func NewPayments() (*Payments, error) {
+	identity, err := provider.NewFullIdentity(ctx, 12, 4)
+	if err != nil {
+		return &Payments{}, ErrIdentity.Wrap(err)
+	}
+	tc := transport.NewClient(identity)
+	conn, err := tc.DialAddress(ctx, port)
```

Comment:
this might be blocked after requiring authorization... so use satellite identity or later private grpc